### PR TITLE
dev-2.1 docs update - overview, examples, programming guide

### PR DESCRIPTION
**Pull request by kkersten**
Further docs restructuring based on VR feedback

- Example applications updated to include all examples, added descriptions, and reorganized from basic to advanced.
- Overview updated to include new architecture diagram and wording.
- Programming guide reorganized to walk through basic concepts and components first.
- Hello Numpy updated to discuss app_common components and how to implement custom code.

### CODE REVIEW

**Comment by kkersten** [comment]
Docs staged here:
https://nvflare-docs-update.readthedocs.io/en/latest/index.html

**Comment by nvkevlu** [comment]
installation.rst is removed, was this intended?

**Comment by kkersten** [comment]
That's right.  I removed it from the navigation in the last set of changes and forgot to remove the source file.  I also updated the examples README.md to point to Quickstart rather than Installation.

**Review by nvkevlu — APPROVED** [review]
Thank you, this looks good.

**Comment by kkersten** [comment]
@nvkevlu One thing that came up when we last discussed is the need to update the links in example_applications.rst for the appropriate branch.  I'm not sure if that should be done now or when dev-2.1 goes live.

**Comment by nvkevlu** [comment]
I created a PR with that: https://github.com/NVIDIA/NVFlare/pull/641

due to the timing of the release though, the docs may not be going live until 2.1.1 is near ready.

File: docs/flare_overview.rst [hunk 1]
```diff
@@ -4,50 +4,89 @@
...
-------------------------------
-We want to enable more people to adopt Federated Learning, whether the user is
+High-level System Architecture
+==============================
+As outlined above, NVIDIA FLARE includes components that allow researchers and developers to build and deploy end-to-end federated learning applications.  The high-level architecture is shown in the diagram below.  This includes the foundational components of the NVIDIA FLARE API and tools for Privacy Preservation and Secure Management of the platform.  On top of this foundation are the building blocks for federated learning applications, with a set of Federation Workflows and Learning Algorithms.
 
-  * a machine learning researcher -- interested in experimenting the latest FL algorithms, or
-  * a data scientist -- interested in applying FL to a real world use case, or
-  * a system integrator -- interested in building a platform to enable Federated Learning for others.
+Alongside this central stack are tools that allow experimentation and proof-of-concept development with the FLARE POC mode and FL Simulator, along with a set of tools used to deploy and manage production workflows.
```
Comment (chesterxgchen) [inline docs/flare_overview.rst:36]: If you don't consider POC as one type of FL simulator, then we have to remove FL simulator. As we don't have it yet.

Comment (kkersten) [reply docs/flare_overview.rst:36]: I do consider POC the current implementation of the FL simulator, just wasn't sure how to word it.  Maybe better to say "Alongside this central stack are tools that allow experimentation and proof-of-concept development with the FL Simulator (POC mode), along with a set of tools used to deploy and manage production workflows."

**Review by chesterxgchen — COMMENTED** [review]
1. notice in FL_stack.png ( FedSM) should be removed, FedSM  is not yet released

File: docs/flare_overview.rst (hunk 1 above)
Comment (chesterxgchen) [reply docs/flare_overview.rst]: that looks good.

**Review by chesterxgchen — APPROVED** [review]
LGTM

**Pull request merged**